### ISSUE TITLE
Allow additional markdown-it plugins

### ISSUE THREAD

**Issue by sqrrrl** [npm/marky-markdown#247]
Any thought to allowing the registration of additional markdown-it plugins? Specifically, I'd like to add support for attributes, but there doesn't seem to be a good way to register additional plugins. Would be nice to be able to pass in additional plugins when parsing, like:

``` javascript
marky('blah blah', { plugins: [ require('markdown-it-attrs')] })
```

If open to it, can submit a pull request.

**Comment by revin** [comment]
Hi @sqrrrl thanks for filing!

On one hand I do like the idea, but on the other hand it feels sort of weird to be exposing markdown-it, which to date has been treated as an internal implementation detail. Adding this capability would tie our version numbering to markdown-it's versioning a bit too, since then we have to track plugin compatibility when markdown-it updates. _Plus_, it's not hard to argue that adding plugins is outside the scope of what marky's trying to do, which is to replicate GitHub's markdown processing (so if there's a plugin that gets us closer to GH rendering, we should just adopt it).

*However*, It's seems pretty useful though, like I could even imagine using it during testing/development of new marky features.

@ashleygwilliams what are your thoughts on adding "secret"/likely-to-break APIs, and explicitly opting to never worry about backwards compatibility?

**Comment by jdarling** [comment]
I just ran into the need to add custom tags to my document, it would be great to either support existing markdown-it plugins or some type of marky plugin for extension.  Personally I think a custom marky style plugin would be a better choice, nothing that says someone couldn't then create a plugin to "port" mardown-it style plugins to marky.

**Comment by revin** [comment]
@sqrrrl we just merged #285, which should allow you do do what you need (see a code example in the PR description). It won't be available in the public package until the next release, but if you clone from master you can check it out.

I'm going to close this now, but if this isn't sufficient for whatever reason, please feel free to reopen.

**Comment by revin** [comment]
oh hmm I never got a notification for your comment @jdarling, sorry about that. I'm not _completely_ clear on what you want; does the `getParser` API give you what you need? If not, maybe open a new issue covering it? Thanks!
